packages/inter-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -1040,11 +1038,15 @@ export const prepareVaultManagerKit = (
             prioritizedVaults.addVault(vaultId, vault);
           }
 
-          // totalCollateral += vault's collateral delta (post — pre)
+          // total += vault's delta (post — pre)
           state.totalCollateral = AmountMath.subtract(
             AmountMath.add(state.totalCollateral, vault.getCollateralAmount()),
             oldCollateral,
           );
+          state.totalDebt = AmountMath.subtract(
+            AmountMath.add(state.totalDebt, vault.getCurrentDebt()),
+            oldDebtNormalized,
+          );
           // debt accounting managed through minting and burning
```

Comment:
this comment is no longer applicable. minting and burning were adjusting, but not publishing. so we do it here to cover both.